### ISSUE TITLE
cluster driver nil error when removing cluster driver of active cluster

### ISSUE THREAD

**Issue by westlywright** [rancher/rancher#18460]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Add a cluster driver via the UI (you can use the instructions [here](https://github.com/rancher/ui-cluster-driver-skel/tree/google-gke-example) to add the google gke example)
Launch a cluster with the `example` cluster driver
At any point after the cluster has been added, i've repeated while the cluster errored out and never finished provisioning and a fully provisioned cluster, remove the cluster driver from via the UI (Drivers -> Cluster)
Refresh the page

**Result:**
The UI won't load and gives the following error
`got unknown driver from kontainer-engine: <nil>`

The rancher server install becomes unrecoverable as far as I can tell. 

**Other details that may be helpful:**
If you look through the network log you can see that its the call to `/v3/clusters` that fails. Container logs confirm the error is happening on on the container not in the UI.

**Comment by daxmc99** [comment]
### To test:
Use the Chrome browser to add a custom cluster driver via the UI. ie
```
https://github.com/rancher/kontainer-engine-driver-example/releases/download/v0.2.3/kontainer-engine-driver-example-darwin-amd64
```
and UI  `http://localhost:3000/component.js`
Follow steps above to observe instability

### To validate:
Attempt to do above
- Option to remove cluster driver via the UI should be greyed out
- Attempting to remove the cluster driver via the API should return a 405 error
See also https://github.com/rancher/rancher/pull/18565/commits/ca17119a302e44487fb1db0414fa7959d4c35c24#diff-b1d73c1a052bd3f017ee05cc461bc8e8R68

**Comment by davidnuzik** [comment]
Version: Master 2.2 (3/9/19) 365fde6ba442
**This is working but I must refresh the page in order to see the Delete button grayed out / delete removed from function menu.** I also must refresh the page after the cluster has been deleted in order to get my Delete button / delete option on menu back:

Screenshot shows the issue. My cluster has been deleted (fully) for a while and I still don't have the delete button / delete function menu item:
![image](https://user-images.githubusercontent.com/45179589/54076454-0c7c9000-4269-11e9-90dd-dbdb0f87d4b4.png)

Otherwise here are my testing notes. Everything worked correctly except I should not have to refresh the page ideally to see the UI change.

I replicated in v2.2.0-rc3 and then upgraded to latest master. Followed the steps and in master I cannot reproduce the issue. When I test either via UI (Delete button) or in the API I cannot delete the kontainer-driver.

UI shows the button is grayed out:
![image](https://user-images.githubusercontent.com/45179589/54076419-b1e33400-4268-11e9-8f3b-5c7af04f313a.png)

UI does not have a delete action on the function menu for the driver:
![image](https://user-images.githubusercontent.com/45179589/54076431-c1627d00-4268-11e9-8f4d-91eb37755104.png)

API returns HTTP 405 with "MethodNotAllow" with message "cluster Driver is in use by one or more clusters, delete clusters before deleting this driver".

I made sure I can still successfully delete the driver both before having any cluster and after deleting the cluster. (But have to refresh page as mentioned)
